### ISSUE TITLE
About SfM-wo-bg and SfM-w-bg

### ISSUE THREAD

**Issue by LiuXD1011** [omniobject3d/OmniObject3D#44]
Hello,

I would like to ask, before novel view synthesis, should I use COLMAP first and then remove the background from the images, or should I remove the background from the images first and then use COLMAP? Or are there other steps to achieve this effect? Thank you for your answer!

the original：
“The SfM-wo-bg and SfM-w-bg settings use images sampled from iPhone videos and camera parameters generated by COLMAP. The difference between them is whether the background is included.”

**Comment by omniobject3d** [comment]
Hi, you can just use the camera poses provided by our datasets, which would apply for both the w/ and w/o mask images. If you want to create your own camera poses via SfM, I would recommend to use COLMAP first and then remove the background to assure more information left in the scene for feature matching during SfM.

**Comment by LiuXD1011** [comment]
Thank you for your response！！
I used the mask from the dataset you provided to remove the background from the images in the images and images_8 folders, and then used the corresponding pose for novel view synthesis.The library I am using is : https://github.com/yenchenlin/nerf-pytorch. The training parameters are all default.
But the images I rendered are all black, I don't know why. May I ask if you can give me some tips and guidance?
Thank you very much!
Here is the file I have processed:
![aaaB](https://github.com/user-attachments/assets/c3277bea-5161-49cd-8aeb-5506edf924f0)

![QQ_1722335519393](https://github.com/user-attachments/assets/7fb95f9c-bb55-46d1-ad5e-27641d64763c)

![QQ_1722335532328](https://github.com/user-attachments/assets/4ec5f3e2-c384-4949-8cd3-d36becd6e414)


![11111111111](https://github.com/user-attachments/assets/33749c77-dd92-468c-8bf3-20c4c0e90f84)

**Comment by wutong16** [comment]
Hi, we need to first figure out whether this is caused by an incorrect camera coordinate system or the overfitting to background color. 

Could you please try to use the same code and camera poses and test it on the original images, where the background is not removed? If it works well, then the previous issue is probably due to the overfitting to the black background during training since the background area is too large. Try to use the mask and re-weight the MSE loss scale of the background, for example, loss_within_mask + \lambda * loss_outside_mask, where you can try 0.1~0.5 for \lambda

**Comment by LiuXD1011** [comment]
The cause of this problem is indeed overfitting. Thank you for your suggestion, and the issue has now been resolved.
Thank you  very much！
